src/features/user/hooks/use-edit-post-card.ts

```diff
@@ -0,0 +1,82 @@
+import { uploadImage } from "../../../api/upload-post-image";
+import { DEFAULT_POST_ERRORS } from "../../../shared/constants/post-content";
+import { postContentEditSchema } from "../../../shared/utils/validate-post";
+import { fromZodError } from "zod-validation-error";
+import { EditPostContent } from "../types/edit-post-content";
+import { PostContent } from "../../../shared/types/post-content";
+import { useUpdatePost } from "../api/use-update-post";
+import { useAddTagToPost } from "../../../api/use-add-tag-to-post";
+import { useDeleteTagPost } from "../api/use-delete-tag-post";
+import { useParams } from "react-router-dom";
+import { useFetchPost } from "../../../api/use-fetch-post";
+
+import { useEffect, useState } from "react";
+
+export const useEditPostCard = (tagsId: number[]) => {
+    const [content, setContent] = useState<string>('');
+    const [image, setImage] = useState<any>(null);
+    const [error, setError] = useState<PostContent>(DEFAULT_POST_ERRORS);
+
+    const {id} = useParams<{id: string | undefined}>()
+
+    const {data: post, isLoading: isLoadingPost} = useFetchPost(id || "")
+    const {mutate: updatePost} = useUpdatePost();
+    const {mutate: addTagToPost} = useAddTagToPost();
+    const {mutate: deleteTagPost} = useDeleteTagPost();
+
```

Comment:
to powinno ci dać do myślenia że heej coś jest nie tak z moim api, skoro do edycji postu, zwykłego szczurowatego POSTU potrzebujesz 3 endpointów